src/util.jl

```diff
@@ -121,8 +121,8 @@ using StaticArrays
 Location in space of the cell at CartesianIndex `I` at face `i`.
 Using `i=0` returns the cell center s.t. `loc = I`.
 """
-@inline loc(i,I::CartesianIndex{N},T=Float64) where N = SVector{N,T}(I.I .- 1.5 .- 0.5 .* δ(i,I).I)
-@inline loc(Ii::CartesianIndex,T=Float64) = loc(last(Ii),Base.front(Ii),T)
+@inline loc(i,I::CartesianIndex{N}) where N = SVector{N}(I.I .- 3//2 .- 1//2 .* δ(i,I).I)
```

Comment:
Have you checked this works on GPU? It was not working for me on the AD branch.

Reply:
No, I have not

Reply:
Ok, I probably won't. We are currently using `1.5f0`... because otherwise it doesn't work in GPUs.

Reply:
I will revert to the type conversion of the SVector for now. We can sort this out later.